modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaGetLayoutDisplayPageProviderCheck.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaGetLayoutDisplayPageProviderCheck
+	extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> importNames = javaClass.getImportNames();
+
+		if (!importNames.contains(
+				"com.liferay.layout.display.page.LayoutDisplayPageProvider")) {
+
+			return content;
+		}
+
+		return _replaceGetLayoutDisplayPageProvider(content, javaClass);
+	}
+
+	private boolean _isLayoutDisplayPageProviderRegistry(
```

Comment:
@michaelmdelcavalcanti, why did you decide to name this method `_isLayoutDisplayPageProviderRegistry` instead of `_isLayoutDisplayPageProvider` or `_hasVariableLayoutDisplayPageProvider`?

I'm asking it because I didn't realize where the name `registry` is coming from. :thinking: